source/slang/hlsl.meta.slang

```diff
@@ -2737,7 +2737,14 @@ matrix<T,N,M> pow(matrix<T,N,M> x, matrix<T,N,M> y)
 
 // Output message
 
-// void printf( string format, ... );
+// __target_intrinsic(cpp, "printf(\"$0\", $1)")
+// __target_intrinsic(GL_EXT_debug_printf, "printf(\"$0\", $1)")
+// __target_intrinsic(GL_EXT_debug_printf)
+
+__target_intrinsic(hlsl)
+__glsl_extension(GL_EXT_debug_printf)
+__target_intrinsic(glsl, "debugPrintfEXT(\"$0\", $1)")
+void printf(String format, float arg0);
```

Comment:
Should use a generic, and perhaps use `NativeString`. No escape is necessary for $0.

```
__target_intrinsic(hlsl)
__glsl_extension(GL_EXT_debug_printf)
__target_intrinsic(glsl, "debugPrintfEXT($0)")
void printf(NativeString format);

__target_intrinsic(hlsl)
__glsl_extension(GL_EXT_debug_printf)
__target_intrinsic(glsl, "debugPrintfEXT($0, $1)")
void printf<T>(NativeString format, T arg0);

__target_intrinsic(hlsl)
__glsl_extension(GL_EXT_debug_printf)
__target_intrinsic(glsl, "debugPrintfEXT($0, $1, $2)")
void printf<T1, T2>(NativeString format, T1 arg0, T2 arg1);

...
```

Reply:
I already tried with NatvieString and it gives similar results.
Your code above produces:
- in glsl output:  `debugPrintfEXT((312235595), (outColor_0.x));`
- spv-asm output: `ExtInst 35(NonSemantic.DebugPrintf) 1(DebugPrintf) 29 34` where `29:     25(int) Constant 312235595`

The value of NativeString gets mapped to an integer (probably an address or some kind of identifier?). I am looking for a way to map a string literal.

Reply:
Then that must be the emit logic for a string literal is missing in `slang-emit-c-like.cpp` and somehow the default path just treats it as a int.

Reply:
slang-emit-glsl.cpp line 1630 is where you need to modify.

Reply:
I tried with what you proposed and it helped. Thank you.